engines/nancy/action/actionmanager.cpp

```diff
@@ -0,0 +1,466 @@
+                    }
+                }
+
+                break;
+            }
+        }
+    }
+}
+
+bool ActionManager::addNewActionRecord(Common::SeekableReadStream &inputData) {
+    inputData.seek(0x30);
+    byte ARType = inputData.readByte();
+    ActionRecord *newRecord = createActionRecord(ARType);
+
+    inputData.seek(0);
+    char *descBuf = new char[0x30];
+    inputData.read(descBuf, 0x30);
+    newRecord->description = Common::String(descBuf);
+    delete[] descBuf;
+
+    newRecord->type = inputData.readByte(); // redundant
+    newRecord->execType = (ActionRecord::ExecutionType)inputData.readByte();
+
+    uint16 localChunkSize = inputData.pos();
+    newRecord->readData(inputData);
+    localChunkSize = inputData.pos() - localChunkSize;
+    localChunkSize += 0x32;
+
+    // If the localChunkSize is less than the total data, there must be dependencies at the end of the chunk
+    uint16 depsDataSize = (uint16)inputData.size() - localChunkSize;
+    if (depsDataSize > 0) {
+        // Each dependency is 0x0C bytes long (in v1)
+        uint numDependencies = depsDataSize / 0xC;
+        if (depsDataSize % 0xC) {
+            error("Action record type %s has incorrect read size", newRecord->getRecordTypeName().c_str());;
+        }
+
+        // Initialize the dependencies data
+        inputData.seek(localChunkSize);
+        for (uint16 i = 0; i < numDependencies; ++i) {
+            newRecord->dependencies.push_back(DependencyRecord());
+            DependencyRecord &dep = newRecord->dependencies.back();
+
+            dep.type = (DependencyType)inputData.readByte();
+            dep.label = inputData.readByte();
+            dep.condition = inputData.readByte();
+            dep.orFlag = inputData.readByte();
+            dep.hours = inputData.readSint16LE();
+            dep.minutes = inputData.readSint16LE();
+            dep.seconds = inputData.readSint16LE();
+            dep.milliseconds = inputData.readSint16LE();
+
+            if (dep.type != kSceneCount || dep.hours != -1 || dep.minutes != -1 || dep.seconds != -1) {
+                dep.timeData = ((dep.hours * 60 + dep.minutes) * 60 + dep.seconds) * 1000 + dep.milliseconds;
+            }
+        }
+    } else {
+        // Set new record to active if it doesn't depend on anything
+        newRecord->isActive = true;
+    }
+
+    _records.push_back(newRecord);
+
+    debugC(1, kDebugActionRecord, "Loaded action record %i, type %s, typeID %i, description \"%s\", execType == %s",
+            _records.size() - 1,
+            newRecord->getRecordTypeName().c_str(),
+            newRecord->type,
+            newRecord->description.c_str(),
+            newRecord->execType == ActionRecord::kRepeating ? "kRepeating" : "kOneShot");
+    for (uint i = 0; i < newRecord->dependencies.size(); ++i) {
+        debugCN(1, kDebugActionRecord, "\tDependency %i: type ", i);
+        switch (newRecord->dependencies[i].type) {
```

Comment:
Imo a local variable for `newRecord->dependencies[i]` would improve the readability of the code a little bit